### PR TITLE
[doc] Contribute a shape to improve resizing support for list node

### PR DESCRIPTION
**Pull request by frouene**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [x] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Until now, when a user resizes a list node, it is not functional.
+Only the container rendering is displayed during the resize action, the sub-container being stretched during the next layout.
+And it's not possible to reduce the size of the container (blocked by the size of the sub-elements).
+Sub-container can be resized in any direction, though it is up to the container to guarantee the width.
+
+== Key Result
+
+* When a container is resized, its sub-container must retain the rendering during the entire action.
+* Sub-container must be resizable vertically only.
```
Comment (sbegaudeau) [inline doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc]: Considering only vertically aligned list layout, there are two use cases:

- The child nodes should fill the parent node
- The child nodes should not fill the parent node

Both behavior are required by our end users. Think about it this way, let's consider the following screenshot from the Sirius Desktop gallery:



Consider the nodes with the blue ribbon on the right of the screenshot. You have a node with a label, for example the "StiffenedGas" one containing three child nodes:

- One with the label "StiffenedGas"
- One with the "limit_tension: Real"
- One with the overrides

Those three child nodes are filling their containing node using a list layout. Yet Inside of the second one, we have a list layout with a single element "limit_tension" which is not filling its containing nodes. In your animated gif, you can find this distinction between the parent node and the yellow node for example.

We cannot guess this by ourselves, the specifier must tell us and we must treat both use cases differently.

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+== Solution
+
+* When a resize event for a list node is triggered, create corresponding event to resize sub-elements.
+* Change sub-container node resizer to only allow vertical resize.
```
Comment (sbegaudeau) [inline doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:68]: - Should the node A1 be vertically resizable?
- If I create 10 nodes in the blue node containing B, should the parent node be resized or should sibling nodes change their size?

We need to specify the various rules.

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+== Scenario
+
+* Container resizing:
+
+image:images/list_node/container_resizing.gif[Container resizing]
```
Comment (sbegaudeau) [inline doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:75]: - Why is the third node with the free form layout resized?
- Why not all three nodes?

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+* Sub-container resizing:
+
+image:images/list_node/sub_container_resizing.gif[Sub-container resizing]
```
Comment (sbegaudeau) [inline doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:79]: Why isn't the blue getting smaller here? Why is the third one changing size?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc (hunk 2 above)
Comment (frouene) [reply doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:68]: It's now more explicit with the CSS rule.
If the blue compartment grows, it depends on the growable childs defined and the space left on the others compartment

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc (hunk 3 above)
Comment (frouene) [reply doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:75]: It's now up to the specifier to decide wich one (or no one) childs are growable

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc (hunk 4 above)
Comment (frouene) [reply doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc:79]: The specifier can now decide this behavior

File: doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc (hunk 1 above)
Comment (frouene) [reply doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc]: done

**Review by AxelRICHARD — APPROVED** [review]

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**